File: domain/src/test/java/mixter/domain/SpyEventPublisher.java
package mixter.domain;

import mixter.domain.core.message.events.MessageQuacked;
import org.assertj.core.util.Lists;

import java.util.List;

public class SpyEventPublisher implements EventPublisher{
    public List<MessageQuacked> publishedEvents= Lists.newArrayList();

    @Override
    public void publish(MessageQuacked messageQuacked) {
        publishedEvents.add(messageQuacked);
    }
}

File: domain/src/test/scala/mixter/domain/SpyEventPublisher.scala
package mixter.domain

import org.scalatest.{Suite, SuiteMixin}

class SpyEventPublisher extends EventPublisher {
  var publishedEvents:Seq[Event]=Seq.empty
  override def publish[T<:Event](event:T): Unit = {
    publishedEvents = publishedEvents :+ event
  }
}
trait SpyEventPublisherFixture extends SuiteMixin { this: Suite =>
  def withSpyEventPublisher(test: SpyEventPublisher=>Any): Any = {
    val spyEventPublisher= new SpyEventPublisher()
    test(spyEventPublisher)
  }
}